### PR TITLE
KAFKA-16584 Make log processing summary configurable or debug

### PR DESCRIPTION
**Pull request by dujian0068**
[KAFKA-16584 Make log processing summary configurable or debug](https://issues.apache.org/jira/browse/KAFKA-16584)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mjsax** [comment]
Please don't forget to update the docs (upgrade guide and configs sections).

**Comment by mjsax** [comment]
@dujian0068 -- you should send a final email to the VOTE thread calling the vote closed and accepted (also summarizing the vote result), and update the KIP wiki pages accordingly. Thanks.

**Comment by dujian0068** [comment]
> Please don't forget to update the docs (upgrade guide and configs sections).

@mjsax Thank you review:
Do you mean the documentation that needs to be updated is in the https://github.com/apache/kafka-site project?

**Comment by mjsax** [comment]
No. The doc changes just go into `kafka.git` as part of this PR -> https://github.com/apache/kafka/tree/trunk/docs/streams

`kafka-site.git` will only be updated during the next release -- you don't have to worry about this part.

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 1]
```diff
@@ -834,6 +834,10 @@ public class StreamsConfig extends AbstractConfig {
     private static final String TASK_ASSIGNOR_CLASS_DOC = "A task assignor class or class name implementing the <code>" +
                                                           TaskAssignor.class.getName() + "</code> interface. Defaults to the <code>HighAvailabilityTaskAssignor</code> class.";
 
+    public static final String LOG_SUMMARY_INTERVAL_MS_CONFIG = "log.summary.interval.ms";
+    private static final String LOG_SUMMARY_INTERVAL_MS_DOC = "The configure control the output interval of summary information.\n" +
+            "if the configuration greater or equal to 0, summary log will be output according to the set time interval;\n" +
+            "If the configuration item value is less than 0, summary output  will be disabled.";
```
Comment (bbejeck) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]:
```suggestion
    private static final String LOG_SUMMARY_INTERVAL_MS_DOC = "This configuration controls the output interval for summary information.\n" +
            "if greater than or equal to 0, the summary log will be output according to the set time interval;\n" +
            "If less than 0, summary output  is disabled.";
```

**Review by bbejeck — COMMENTED** [review]
Thanks for the KIP and contribution @dujian0068!

Overall this looks good, I have one minor comment.

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java (hunk 1 above)
Comment (dujian0068) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: Thank you 
I have changed the DOC

**Comment by dujian0068** [comment]
Thank you @mjsax 
I have updated config-stream doc. Do I need to update upgrade-guide? Can you tell me in which cases I need to update upgrade-guide?

**Comment by bbejeck** [comment]
Failures unrelated

```
[Build / JDK 21 and Scala 2.13 / kafka.server.DelegationTokenRequestsTest."testDelegationTokenRequests(String).quorum=kraft"](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-16509/3/testReport/junit/kafka.server/DelegationTokenRequestsTest/Build___JDK_21_and_Scala_2_13____testDelegationTokenRequests_String__quorum_kraft_/)
[Build / JDK 21 and Scala 2.13 / kafka.zk.ZkMigrationFailoverTest.testControllerFailoverZkRace()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-16509/3/testReport/junit/kafka.zk/ZkMigrationFailoverTest/Build___JDK_21_and_Scala_2_13___testControllerFailoverZkRace__/)
[Build / JDK 21 and Scala 2.13 / org.apache.kafka.controller.QuorumControllerTest.testFenceMultipleBrokers()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-16509/3/testReport/junit/org.apache.kafka.controller/QuorumControllerTest/Build___JDK_21_and_Scala_2_13___testFenceMultipleBrokers__/)
[Build / JDK 21 and Scala 2.13 / org.apache.kafka.controller.QuorumControllerTest.testDelayedConfigurationOperations()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-16509/3/testReport/junit/org.apache.kafka.controller/QuorumControllerTest/Build___JDK_21_and_Scala_2_13___testDelayedConfigurationOperations__/)
[Build / JDK 8 and Scala 2.12 / org.apache.kafka.connect.mirror.integration.MirrorConnectorsIntegrationTransactionsTest.testSyncTopicConfigs()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-16509/3/testReport/junit/org.apache.kafka.connect.mirror.integration/MirrorConnectorsIntegrationTransactionsTest/Build___JDK_8_and_Scala_2_12___testSyncTopicConfigs__/)
[Build / JDK 8 and Scala 2.12 / kafka.api.ConsumerBounceTest.testConsumptionWithBrokerFailures()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-16509/3/testReport/junit/kafka.api/ConsumerBounceTest/Build___JDK_8_and_Scala_2_12___testConsumptionWithBrokerFailures__/)
[Build / JDK 8 and Scala 2.12 / org.apache.kafka.tools.GetOffsetShellTest.testTopicPartitionsArg [2] Type=Raft-Isolated, MetadataVersion=4.0-IV0,Security=PLAINTEXT](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-16509/3/testReport/junit/org.apache.kafka.tools/GetOffsetShellTest/Build___JDK_8_and_Scala_2_12___testTopicPartitionsArg__2__Type_Raft_Isolated__MetadataVersion_4_0_IV0_Security_PLAINTEXT/)
[Build / JDK 8 and Scala 2.12 / org.apache.kafka.tools.MetadataQuorumCommandTest.testDescribeQuorumReplicationSuccessful [2] Type=Raft-Combined, MetadataVersion=4.0-IV0,Security=PLAINTEXT](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-16509/3/testReport/junit/org.apache.kafka.tools/MetadataQuorumCommandTest/Build___JDK_8_and_Scala_2_12___testDescribeQuorumReplicationSuccessful__2__Type_Raft_Combined__MetadataVersion_4_0_IV0_Security_PLAINTEXT/)
```

**Review by bbejeck — APPROVED** [review]
LGTM

**Comment by bbejeck** [comment]
merged #16509 into trunk

**Comment by bbejeck** [comment]
Thanks @dujian0068 for the contribution!

**Comment by bbejeck** [comment]
@dujian0068 - can you also add an entry in the `docs/streams/upgrade-guide.html` with the changes your KIP introduced?

**Comment by dujian0068** [comment]
> @dujian0068 - can you also add an entry in the `docs/streams/upgrade-guide.html` with the changes your KIP introduced?

@bbejeck Of course 
I have submitted a new merge request：https://github.com/apache/kafka/pull/16709
Please help me review it

I considered updating this file before, but I saw that it was full of major updates, and my submission was a very small optimization, so I did not update this file. Can I ask which updates require updating this file?

**Pull request merged**